webextensions/browser-compat-data.json

```diff
@@ -1199,6 +1199,9 @@
     }, 
     "TemplateType": {
       "Chrome": {
+        "notes": [
+          "For the 'basic' type, 'iconUrl' is mandatory."
+        ],
```

Comment:
Thanks for pointing out that `iconUrl` was mentioned as mandatory in [NotificationOptions](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/notifications/NotificationOptions). I had missed that it was stated there. Unfortunately, the information in that page was erroneous. For Firefox, the `iconUrl` property is optional in `notifications.create()` (based on testing  and [source code](https://dxr.mozilla.org/mozilla-central/source/toolkit/components/extensions/schemas/notifications.json#51)). I have update that page to reflect testing and source code (i.e. not sate that `iconUrl` is mandatory in `notifications.create()`).  I corrected the [NotificationOptions](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/notifications/NotificationOptions) page based on testing and the source code shortly after I saw your comment.

Where to document all of the requirements for the various notification types and methods is a good question. The if a property is mandatory or optional depends on both the API method being called and the `type` of notification.  While the situation is significantly simplified at the moment with Firefox not currently supporting `notifications.update()` and only supporting `'basic'` for the `type`, the way that these are displayed to the add-on developer requires them to drill down from the API pages to all of the Type pages. Particularly for a developer who is porting from Chrome, the need to drill down to each Type page is not, necessarily, obvious.

It does bring up a couple of questions:

1. Where should the mandatory/optional nature of the properties be documented:
 - In `notifications.create()` and `notifications.update()`?  There are differences between mandatory/optional between that method and `notifications.update()` (which is not yet supported in Firefox). Which properties are required could be included in the documentation for each method. At the moment, this is probably the easiest as only `notifications.create()` is supported and only for the `'basic'` type. However, if it is in the method (which is where programmers are probably going to look first; thus it is good for it to be there), then we need to consider that this will have to cover what is mandatory/required for each `type` when additional types are supported. When that happens, how it is documented could change, but is more likely to organically grow from whatever is used now.
 - In NotificationOptions?  There are mandatory/optional differences based on both the method in which the NotificationOptions object is used and, in the future, based on the TemplateType.
 - In [TemplateType](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/Notifications/TemplateType)?  This page already has a breakdown of what properties are available for the different types. It would need to also have which are mandatory based on the method being called.
2. Should such things be documented in more than place? My expectation is that if the amount of documentation was significant then this should be a no (but that a pointer or summary should be provided where it might be needed). However, currently the information is a limited subset of what it is expected to be when more types are supported and/or when `notifications.update()` is supported.

Where that information is documented is the obvious page on which to have notes with the compatibility differences between Firefox and Chrome (i.e. what is dealt with in this repository).